src/message/header/content_type.rs

```diff
@@ -0,0 +1,129 @@
+use std::{
+    error::Error as StdError,
+    fmt::{self, Display, Result as FmtResult},
+    str::{from_utf8, FromStr},
+};
+
+use hyperx::{
+    header::{Formatter as HeaderFormatter, Header, RawLike},
+    Error as HeaderError, Result as HyperResult,
+};
+use mime::Mime;
+
+/// `Content-Type` of the body
+///
+/// Defined in [RFC2045](https://tools.ietf.org/html/rfc2045#section-5)
+#[derive(Debug, Clone, PartialEq)]
+pub struct ContentType(Mime);
+
+impl ContentType {
+    /// A `ContentType` of type `text/plain; charset=utf-8`
+    pub const PLAIN_STRING: ContentType = Self::from_mime(mime::TEXT_PLAIN_UTF_8);
```

Comment:
I wonder if we shouldn't keep something closer to the actual mime type, like `TEXT_PLAIN` as it is a pretty standard and well known type (not sure about the `UTF_8` part, if our API uses Rust strings we probably don't need to mention it indeed).

Reply:
Sounds good. Thinking more about it almost nobody is going to use a non-utf8 string for the plaintext or the html version of the body, so `TEXT_PLAIN` and `TEXT_HTML` with a comment explaining that they are expected to be utf8 makes sense here.